src/nm-checkpoint.c

```diff
@@ -245,6 +248,15 @@ restore_and_activate_connection (NMCheckpoint *self,
 			g_clear_error (&local_error);
 			return FALSE;
 		}
+
+		/* If the device is software, a brand new NMDevice may have been created */
+		if (   dev_checkpoint->is_software
+		    && !dev_checkpoint->device) {
+			dev_checkpoint->device = nm_manager_get_device (priv->manager,
+			                                                dev_checkpoint->original_dev_name,
+			                                                dev_checkpoint->dev_type);
+			g_object_ref (dev_checkpoint->device);
```

Comment:
you need to handle the possibility that `nm_manager_get_device()` won't find the device.

that may just mean to use `nm_g_object_ref()` instead of `g_object_ref()`.

Reply:
done